experimental/adaptive-dialog/docs/recognizers-rules-steps-reference.md

```diff
@@ -0,0 +1,739 @@
+// a confirmation input to the user.
+ConfirmationDialog.AddRule(new EventRule(["Contoso.travelBot.confirm"], 
+    steps: new List<IDialog> {
+        // Add confirmation input.
+        new ConfirmInput()
+        {
+            Property = "turn.contoso.travelBot.confirmOutcome",
+            // Since this prompt is built as a generic confirmation wrapper, the actual prompt text is 
+            // read from a specific memory location. The caller of this dialog needs to set the prompt
+            // string to that location before calling the "ConfirmationDialog".
+            // All prompts support rich language generation based resolution for output generation. 
+            // See ../../language-generation/README.md to learn more.
+            Prompt = new ActivityTemplate("{turn.contoso.travelBot.confirmPromptMessage}")
+        }
+    });
+```
+
+### NumberInput
+
+Asks for a number.
+
+**Note:** By default, ConfirmInput does not trigger consultation if the confirm input recognizer has a high confidence match against provided response. 
+
+| Property         | Description                                                                        |
+|------------------|------------------------------------------------------------------------------------|
+| property         | Property this input dialog is bound to                                             |
+| prompt           | Initial prompt response to ask for user input                                      |
+| retryPrompt      | Response on retry                                                                  |
+| invalidPrompt    | Response when input is not recognized or not valid for the expected input type     |
+| minValue         | The min value which is valid                                                       |
+| maxValue         | The max value which is valid                                                       |
+
+
+``` C#
+// Create adaptive dialog.
+var getNumberDialog = new AdaptiveDialog("ConfirmationDialog");
+// We are using an Event rule here so any other step can just call EmitEvent when it wants to collect
+// a number from the user.
+getNumberDialog.AddRule(new EventRule(["Contoso.travelBot.getNumber"], 
+    steps: new List<IDialog> {
+        new NumberInput<int>()
+        {
+            Property = "turn.contoso.travelBot.numberPrompt",
+            Prompt = new ActivityTemplate("{turn.contoso.travelBot.numberMessage}")
+        }
+    });
+```
+
+## Steps
+_Steps_ help put together the flow of conversation when a specific event is captured via a Rule. **_Note:_** unlike Waterfall dialog where each step is a function, each step in an Adaptive dialog is in itself a dialog. This enables adaptive dialogs by design to 
+- have a much cleaner ability to handle and deal with interruptions.
+- branch conditionally based on context or state.
+
+Adaptive dialogs support the following steps - 
```

Comment:
We probably need to explain why we even need steps, or to be specific why do we need to wrap standard 'code' with steps. It is strange to use an IfCondition instead of just writing an if statement. 

If I am a new user to this, i will ask, what is this strange code?